### PR TITLE
Non-Confirming Funders Notification and Delete Styling issues

### PR DESCRIPTION
**Pull request by MSDrao**
### Pull Request Checklist: 
- [x] Positive Test Case Written by Dev

- [x] Automated Testing

- [ ] Sufficient User and Developer Documentation

- [ ] Passing Jenkins Build

- [ ] Peer Code review and approval

### CODE REVIEW

File: theme/static/css/hydroshare_core.css [hunk 1]
```diff
@@ -4560,7 +4560,17 @@ span.caption-file-type {
 
 
 .btn-remove {
-	font-size: 18px;
+	font-size: 14px;
```
Comment (devincowan) [inline theme/static/css/hydroshare_core.css:4563]: be careful altering this class -- it is used in many elements!
If you're going to modify it, I would do a search in the codebase for `btn-remove` to make sure that you're not breaking styles in other places

File: theme/static/js/hs-vue/funding-agency-app.js [hunk 2]
```diff
@@ -56,22 +56,25 @@ let fundingAgenciesApp = new Vue({
         for (let umatch of unmatched) {
           this.unmatchedFunders.push(umatch.funderName);
         }
+    
+        console.log("Unmatched Funders:", this.unmatchedFunders); // Add this to debug
```
Comment (devincowan) [inline theme/static/js/hs-vue/funding-agency-app.js:60]: remove debug log?

File: theme/static/js/hs-vue/funding-agency-app.js [hunk 3]
```diff
@@ -38,9 +38,9 @@ let fundingAgenciesApp = new Vue({
     filteredWords: [],
   },
   mounted() {
-    if (this.selfAccessLevel === "owner" && this.resPublished) {
-      this.checkFunderNamesExistInCrossref(this.fundingAgencies);
-    }
+    if (this.resourceMode === 'Edit' && (this.selfAccessLevel === 'owner' || this.selfAccessLevel === 'editor') && this.resPublished) {
+    this.checkFunderNamesExistInCrossref(this.fundingAgencies);
+  }
```
Comment (devincowan) [inline theme/static/js/hs-vue/funding-agency-app.js:43]: can you remove this? is it handled by L241-246 now?

**Review by devincowan — CHANGES_REQUESTED** [review]
a few questions for you to consider @MSDrao

File: theme/templates/resource-landing-page/funding-agency.html [hunk 4]
```diff
@@ -110,6 +110,21 @@ <h4 class="modal-title" id="delete-funding-agency">Delete Funding Agency</h4>
...
+                          <i class="glyphicon glyphicon-info-sign" style="margin-right: 20px;"></i>
+                          <em style="padding-right:20px;">
+                            This resource contains funders (listed below) that do not exist in the <a href="https://www.crossref.org/services/funder-registry" target="_blank">Open Funder Registry</a>:
+                            <ul>
+                              <li v-for="funder in unmatchedFunders">${funder}</li>
+                            </ul>
+                            We recommend updating the funders to conform to the <a href="https://www.crossref.org/services/funder-registry" target="_blank">Open Funder Registry</a> to ensure consistency and ease of reporting.
+                          </em>
+                        </div>
+                      </div>
```
Comment (devincowan) [inline theme/templates/resource-landing-page/funding-agency.html:127]: why are you including this code block within the `DELETE FUNDING AGENCY MODAL`? This seems like a pretty strange place for it. Can you explain why you are moving it?

File: theme/static/js/hs-vue/funding-agency-app.js [hunk 5]
```diff
@@ -224,14 +224,25 @@ let fundingAgenciesApp = new Vue({
...
       this.$refs.agencyNameInput.inputValue = this.currentlyEditing.agency_name;
+    
+      // Only check funder names when editing the resource
+      if (this.selfAccessLevel === "owner" && this.resPublished) {
+        this.checkFunderNamesExistInCrossref(this.fundingAgencies).then(() => {
+          if (this.unmatchedFunders.length > 0) {
+            this.showFundersAlert(); // Show alert if unmatched funders are found
+          }
+        });
+      }
```
Comment (devincowan) [inline theme/static/js/hs-vue/funding-agency-app.js:245]: this check will only be run when `openEditModal` is called. This is not the same as when the resource is in "edit mode" -> `openEditModal` is a function within this Vue app that is called when one clicks the button to edit a single funder
image

**Review by devincowan — CHANGES_REQUESTED** [review]

File: theme/static/js/hs-vue/funding-agency-app.js (hunk 5 above)
Comment (devincowan) [reply theme/static/js/hs-vue/funding-agency-app.js:245]: we discussed this, marking resolved

File: theme/templates/resource-landing-page/funding-agency.html (hunk 4 above)
Comment (devincowan) [reply theme/templates/resource-landing-page/funding-agency.html:127]: @MSDrao I still think this is pretty funky: showing the notification within the delete modal:
image

https://github.com/user-attachments/assets/08fc177d-0a1d-46b8-a429-02977f6ac933

I think I would still recommend keeping this notification where it was before

Comment (devincowan) [reply theme/templates/resource-landing-page/funding-agency.html:127]: @martinseul what is your opinion? ☝️ (see video to see what we are talking about)

Comment (martinseul) [reply theme/templates/resource-landing-page/funding-agency.html:127]: I'd agree with Devin these are separate task and should not be conflated

**Review by devincowan — APPROVED** [review]

**Comment by hydrocheck** [comment]
We seem to have a regression in adding comments

https://github.com/user-attachments/assets/6056fbc9-57c5-47a1-b9cc-1d6629388663

**Comment by hydrocheck** [comment]
When clicking "Edit" on the Funding Agency, the name is duplicated in the bullet list

Screenshot 2024-11-06 at 12 17 54 AM

**Pull request merged**